service/controller/app/v1/resource_set.go

```diff
@@ -174,6 +188,7 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		secretResource,
 		chartResource,
 		statusResource,
+		kubeconfigFinalizerResource,
```

Comment:
Put this resource on the last so we give chances to other resources to reconciliation with deletion events.